app/src/main/java/com/globant/practice/presentation/view/fragment/WebClientFragment.java

```diff
@@ -0,0 +1,181 @@
+            super.onReceivedError(view, errorCode, description, failingUrl);
+            presenter.webPageError(failingUrl);
+        }
+
+        @TargetApi(android.os.Build.VERSION_CODES.M)
+        @Override
+        public void onReceivedError(WebView view, WebResourceRequest request, WebResourceError error) {
+            onReceivedError(view, error.getErrorCode(), error.getDescription().toString(), request.getUrl().toString());
+        }
+    };
+
+    /**
+     * Returns a new instance if WebClientFragment with the necessary arguments that the fragment needs
+     *
+     * @param htmlUrl    url of the profile or repository address
+     * @param detailType Indicates if the web page is a profile or a repository
+     * @return new instance of WebClientFragment with the necessary arguments
+     */
+    public static WebClientFragment newInstance(String htmlUrl, String detailType) {
+        WebClientFragment fragment = new WebClientFragment();
+        Bundle bundle = new Bundle();
+        bundle.putString(HTML_URL_KEY, htmlUrl);
+        bundle.putString(DETAIL_TYPE_KEY, detailType);
+        fragment.setArguments(bundle);
+        return fragment;
+    }
+
+    /**
+     * Inflates the fragment_web_client and initializes the WebClientPresenter
+     *
+     * @param inflater
+     * @param container
+     * @param savedInstanceState
+     * @return
+     */
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        ((PracticeApplication) getActivity().getApplication()).getApplicationComponent().inject(this);
+        return inflater.inflate(R.layout.fragment_web_client, container, false);
+    }
+
+    /**
+     * Initializes the view items and set the ActionBar title
+     *
+     * @param view
+     * @param savedInstanceState
+     */
+    @Override
+    public void onViewCreated(View view, Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        if (!getArguments().getString(DETAIL_TYPE_KEY).isEmpty()) {
+            ((AppCompatActivity) getActivity()).getSupportActionBar().setTitle(getArguments().getString(DETAIL_TYPE_KEY));
+        }
+        webClientWebView = (WebView) view.findViewById(R.id.web_client_webview);
+        webPageLoadingIndicator = new ProgressDialog(view.getContext());
+        webPageLoadingIndicator.setIndeterminate(true);
+        webPageLoadingIndicator.setProgressStyle(ProgressDialog.STYLE_SPINNER);
+    }
+
+    /**
+     * Attach the view on the presenter and initializes the web page load process
+     */
+    @Override
+    public void onResume() {
+        super.onResume();
+        presenter.attachView(this);
+        presenter.loadWebPage(getArguments().getString(HTML_URL_KEY), getArguments().getString(DETAIL_TYPE_KEY));
+    }
+
+    /**
+     * Detach the view on the presenter
+     */
+    @Override
+    public void onDetach() {
```

Comment:
Generally, presenter.detachView is implemented in *onPause*. Excuse me for not seeing this before.